File: questapp-frontend/src/service/CommentService.js
import axios from "axios";

const base_url = "http://localhost:8080/comments";

class CommentService {
  async getAllComment(postId) {
    return await axios.get(`${base_url}?postId=${postId}`);
  }
  async createOneComment(comment) {
    return await axios.post(base_url, comment);
  }
}

export default new CommentService();

File: questapp-backend/src/main/java/com/example/questappbackend/services/abs/CommentService.java
package com.example.questappbackend.services.abs;

import com.example.questappbackend.dto.CommentDtoForCreate;
import com.example.questappbackend.dto.CommentDtoForUpdate;
import com.example.questappbackend.entities.Comment;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;
import java.util.Optional;

public interface CommentService {

    List<Comment> getAllCommentsWithParam (Optional<Long> userId, Optional<Long> postId);

    Comment getOneComment (Long commentId);

    Comment createOneComment (CommentDtoForCreate request);

    Comment updateOneComment (Long commentId, CommentDtoForUpdate request);

    void deleteOneComment (Long commentId);

}
